### PR TITLE
[Keyboard] Gowla Macro Board

### PR DESCRIPTION
**Pull request by SamGowland**
## Description

New PCB for a simple compact 3x3 Macro Board design for use with QMK and Pro Micro. 

First QMK and PCB project. PCB available to buy, this is detailed in the hardware repo. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gowla/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+[1] = LAYOUT( \
+  KC_ESC,   KC_MPLY, 	RESET, \
+  KC_P7, 	KC_P1, 		M(0), \
+  KC_F1,	KC_F2,  	M(1)  \
+)
+
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) {
```
Comment (fauxpark) [inline keyboards/gowla/keymaps/default/keymap.c:19]: `action_get_macro()` is deprecated: https://docs.qmk.fm/#/feature_macros?id=the-new-way-send_string-amp-process_record_user

File: keyboards/gowla/gowla.c [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include "gowla.h"
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (fauxpark) [inline keyboards/gowla/gowla.c:5]:
```suggestion
```

File: keyboards/gowla/config.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
+#define CONFIG_H
+
```
Comment (fauxpark) [inline keyboards/gowla/config.h:18]:
```suggestion
#pragma once

```

File: keyboards/gowla/config.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0007
+#define DEVICE_VER 		0x0001
```
Comment (fauxpark) [inline keyboards/gowla/config.h:24]:
```suggestion
#define DEVICE_VER      0x0001
```

File: keyboards/gowla/config.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+/* Pin-out */
+#define MATRIX_ROW_PINS { D1, D0, D4 }
+#define MATRIX_COL_PINS { B5, B4, E6 }
+#define UNUSED_PINS
+
+/* ROW2COL */
+#define DIODE_DIRECTION ROW2COL
+
+#endif
```
Comment (fauxpark) [inline keyboards/gowla/config.h:41]:
```suggestion
```

File: keyboards/gowla/gowla.h [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#ifndef GOWLA_H
+#define GOWLA_H
+
```
Comment (fauxpark) [inline keyboards/gowla/gowla.h:3]:
```suggestion
#pragma once

```

File: keyboards/gowla/gowla.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+#include "quantum.h"
+
+#define LAYOUT( \
+      k00, k01, k02, \
+      k10, k11, k12, \
+      k20, k21, k22  \
+) \
+{ \
```
Comment (fauxpark) [inline keyboards/gowla/gowla.h:11]:
```suggestion
) { \
```

File: keyboards/gowla/gowla.h [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    { k00, k01, k02 }, \
+    { k10, k11, k12 }, \
+    { k20, k21, k22 } \
+} 
+
+#endif
```
Comment (fauxpark) [inline keyboards/gowla/gowla.h:17]:
```suggestion
}
```

File: keyboards/gowla/info.json [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  "keyboard_name": "Gowla Macro Pad",
+  "url": "https://github.com/SamGowland/Gowla-Macro-Board",
+  "maintainer": "Gowla",
+  "width": 3,
+  "height": 3,
+  "layouts": {
+    "LAYOUT": {
+      "layout": [{"label":"k00", "x":0, "y":0}, {"label":"k01", "x":1, "y":0}, {"label":"k02", "x":2, "y":0}, {"label":"k10", "x":0, "y":1}, {"label":"k11", "x":1, "y":1}, {"label":"k12", "x":2, "y":1}, {"label":"k20", "x":0, "y":2}, {"label":"k21", "x":1, "y":2}, {"label":"k22", "x":2, "y":2}]
+    }
+  }
```
Comment (fauxpark) [inline keyboards/gowla/info.json:11]:
```suggestion
    "keyboard_name": "Gowla Macro Pad",
    "url": "https://github.com/SamGowland/Gowla-Macro-Board",
    "maintainer": "Gowla",
    "width": 3,
    "height": 3,
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"x": 0, "y": 0},
                {"x": 1, "y": 0},
                {"x": 2, "y": 0},

                {"x": 0, "y": 1},
                {"x": 1, "y": 1},
                {"x": 2, "y": 1},

                {"x": 0, "y": 2},
                {"x": 1, "y": 2},
                {"x": 2, "y": 2}
            ]
        }
    }
```

File: keyboards/gowla/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  KC_VOLD,       		KC_UP,			KC_VOLU, \
+  KC_LEFT,       		KC_DOWN,      	KC_RIGHT \
+),
+
+[1] = LAYOUT( \
+  KC_ESC,   KC_MPLY, 	RESET, \
+  KC_P7, 	KC_P1, 		M(0), \
+  KC_F1,	KC_F2,  	M(1)  \
+)
+
```
Comment (fauxpark) [inline keyboards/gowla/keymaps/default/keymap.c:16]:
```suggestion
    [0] = LAYOUT(
        LT(1, KC_MPRV), KC_MNXT, KC_MPLY,
        KC_VOLD,        KC_UP,   KC_VOLU,
        KC_LEFT,        KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT(
        KC_ESC,  KC_MPLY, RESET,
        KC_P7,   KC_P1,   M(0),
        KC_F1,   KC_F2,   M(1)
    )
```

File: keyboards/gowla/readme.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+Keyboard Maintainer: [Gowla](https://github.com/SamGowland/)
+Hardware Supported: Pro Micro ATmega32U4, Cherry Mount Switches  
+Hardware Availability: [Repo]https://github.com/SamGowland/Gowla-Macro-Board)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make gowla:default
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (fauxpark) [inline keyboards/gowla/readme.md:14]:
```suggestion
# Gowla Macro Board

Simple Compact 3x3 Macro Board for use with QMK, designed and sold by Gowla.

* Keyboard Maintainer: [Gowla](https://github.com/SamGowland/)
* Hardware Supported: Pro Micro ATmega32U4, Cherry Mount Switches
* Hardware Availability: [Repo]https://github.com/SamGowland/Gowla-Macro-Board)

Make example for this keyboard (after setting up your build environment):

    make gowla:default

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/gowla/rules.mk [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = yes
+TAP_DANCE_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/gowla/rules.mk:34]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```

File: keyboards/gowla/gowla.h [hunk 13]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/gowla/gowla.h:5]:
```suggestion
#define LAYOUT_ortho_3x3( \
```

File: keyboards/gowla/info.json [hunk 14]
```diff
@@ -0,0 +1,24 @@
+{
+    "keyboard_name": "Gowla Macro Pad",
+    "url": "https://github.com/SamGowland/Gowla-Macro-Board",
+    "maintainer": "Gowla",
+    "width": 3,
+    "height": 3,
+    "layouts": {
+        "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/gowla/info.json:8]:
```suggestion
        "LAYOUT_ortho_3x3": {
```

File: keyboards/gowla/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,26 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/gowla/keymaps/default/keymap.c:4]:
```suggestion
    [0] = LAYOUT_ortho_3x3(
```

File: keyboards/gowla/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+        LT(1, KC_MPRV), KC_MNXT, KC_MPLY,
+        KC_VOLD,        KC_UP,   KC_VOLU,
+        KC_LEFT,        KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/gowla/keymaps/default/keymap.c:9]:
```suggestion
    [1] = LAYOUT_ortho_3x3(
```

File: keyboards/gowla/rules.mk [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/gowla/rules.mk:29]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
```

File: keyboards/gowla/config.h [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0007
```
Comment (fauxpark) [inline keyboards/gowla/config.h:22]:
```suggestion
#define PRODUCT_ID      0xE9B6
```
```
echo -n "Gowla Macro Pad" | shasum | cut -c1-4
```

File: keyboards/gowla/readme.md [hunk 19]
```diff
@@ -0,0 +1,13 @@
+# Gowla Macro Board
```
Comment (Erovia) [inline keyboards/gowla/readme.md:1]:
```suggestion
# Gowla Macro Board

![Gowla](https://github.com/SamGowland/Gowla-Macro-Board/blob/master/Images/comp.jpg)
```

File: keyboards/gowla/config.h [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
```
Comment (Erovia) [inline keyboards/gowla/config.h:21]: This is just a food for thoughts and not a blocker, but if you intend to add support for more of your designs, I recommend creating a vendor directory and putting the board sources there.

Like `keyboards/gowla/macro_board` or something similar.

Comment (SamGowland) [reply keyboards/gowla/config.h:21]: Thanks for your suggestion. I think I will keep this in mind and make this change once/if I create any new designs. Which might not be for a few weeks :)

**Pull request merged**